### PR TITLE
Issue 58/dataset page

### PR DESCRIPTION
**Pull request by admy7**
Includes several features concerning the dataset page: 
- UI (responsive)
- Data fetching: datasets and filter values
- Filtering
- Pagination

I am going to add unit tests (I will not merge it before, but you can already review all the changes as there are many :p) 

Screenshot 2024-03-03 at 23 17 26
Screenshot 2024-03-03 at 23 18 07
Screenshot 2024-03-03 at 23 18 59
Screenshot 2024-03-03 at 23 19 20

### CODE REVIEW

File: src/app/datasets/page.tsx [hunk 1]
```diff
@@ -2,6 +2,67 @@
...
+import { PackageSearchOptions } from "@/services/ckan/types/packageSearch.types";
+import { redirect } from "next/navigation";
+
+type DatasetPageProps = {
+  searchParams: Record<string, string>;
+};
+
+async function DatasetPage({ searchParams }: DatasetPageProps) {
+  if (!searchParams?.page) {
+    redirect("/datasets?page=1");
```
Comment (brunopacheco1) [inline src/app/datasets/page.tsx:17]: question: I don't page load events in React, but this redirect may trigger twice the server side pagination/search. Can you please check?

**Review by brunopacheco1 — APPROVED** [review]

File: src/utils/textProcessing.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { pixelWidthToScreenSize } from './windowSize';
+
+const SCREEN_SIZE_TO_MAX_WORDS = {
+  XL: 100,
+  LG: 70,
+  MD: 40,
+  SM: 30,
+};
+
+function truncateDescription(description: string, screenWidth: number): string {
```
Comment (inderps) [inline src/utils/textProcessing.ts:14]: we need to have unit test for these kind of code. Maybe add another ticket in backlog if you plan to now finish your PR before your vacation

Comment (admy7) [reply src/utils/textProcessing.ts:14]: Yep I wrote some units tests. Will commit them

File: src/services/ckan/datasetList.ts [hunk 3]
```diff
@@ -23,19 +24,26 @@ export const makeDatasetList = (DMS: string) => {
 };
 
 const buildFilterQueryPart = (filters: string[]): string => {
-  return filters.length > 0 ? `(${filters.join(' OR ')})` : '';
+  return filters.length > 0
+    ? `(${filters
+        .map((filter: string) => (filter.startsWith('http') ? `"${filter}"` : filter))
```
Comment (inderps) [inline src/services/ckan/datasetList.ts:29]: Unit test for this new logic. Let's add a separate ticket in backlog for adding tests for all this logic

This ticket can be picked in future

File: src/utils/windowSize.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+enum SCREEN_SIZE {
+  SM = 'SM',
+  MD = 'MD',
+  LG = 'LG',
+  XL = 'XL',
+}
+
+function pixelWidthToScreenSize(width: number): SCREEN_SIZE {
```
[the remark this answers is not in the document]
Comment (admy7) [reply src/utils/windowSize.ts:12]: Done

File: src/utils/textProcessing.ts (hunk 2 above)
Comment (admy7) [reply src/utils/textProcessing.ts:14]: Done

File: src/services/ckan/datasetList.ts (hunk 3 above)
Comment (admy7) [reply src/services/ckan/datasetList.ts:28]: Unit test has now been implemented

File: src/app/datasets/page.tsx (hunk 1 above)
Comment (admy7) [reply src/app/datasets/page.tsx:18]: Good point, I haven't had much time but I did not find a clear answer. I will write it down to not forget it

**Pull request merged**